### ISSUE TITLE
Support Gradle configuration cache

### ISSUE THREAD

**Issue by Kantis** [skippy-io/skippy#159]
Running `SkippyAnalyze` task in a project which uses Gradle's configuration cache yields the following error:

```
1 problem was found storing the configuration cache.
- Task `:autoinvoice:skippyAnalyze` of type `io.skippy.gradle.SkippyAnalyzeTask`: invocation of 'Task.project' at execution time is unsupported.
```

**Comment by fmck3516** [comment]
Corresponding E2E test: https://github.com/skippy-io/skippy-regression-suite/pull/2

**Comment by fmck3516** [comment]
Quick update: All E2E test except [TestFailureTest.java](https://github.com/skippy-io/skippy-regression-suite/blob/main/src/test/java/io/skippy/test/gradle/TestFailureTest.java) pass with Configuration Cache enabled using the changes in #160 . I'm still investigating why the result property for a failed test is set incorrectly with Configuration Cache enabled.

**Comment by Kantis** [comment]
My best guess is that it's somehow related to not having the class-dirs as inputs, so Gradle doesn't realize the task is stale (due to classpath changing), and the task is not re-run as a result. Simply adding `--rerun-tasks` to the arguments here might be enough to validate the hypothesis. https://github.com/skippy-io/skippy-regression-suite/blob/8192a55ca11e9793d3ce803fc7be892d5e10ceb3/src/test/java/io/skippy/test/gradle/TestFailureTest.java#L40C17-L40C73

**Comment by fmck3516** [comment]
I added some logging. From what I can tell, it's not due to tasks not being executed.

What I see:  `SkippyAnalyzeTask` and the test tasks use different instances of `TestFailedListener`. The tests tasks report failed tests to `TestFailedListener@00000000`. `SkippyAnalyzeTask` references `TestFailedListener@11111111`.

**Comment by fmck3516** [comment]
I will create a minimal project that demonstrates my problem and reach out to the Gradle folks for help.

**Comment by fmck3516** [comment]
FYI: https://discuss.gradle.org/t/sharing-objects-across-taks-with-configuration-cache-enabled/48598
Sample repo to re-produce the issue in isolation: https://github.com/skippy-io/gradle-configuration-cache-issue

**Comment by fmck3516** [comment]
Accidentally closed this issue when squashing the PR. Have to add additional tests and figure out how to solve that for Gradle 7.

@Kantis What version of Gradle are you using? This should be fixed in 0.0.21-SNAPSHOT if you use Gradle 8 or above.

**Comment by fmck3516** [comment]
You can use snapshot versions as follows:
```
buildscript {
    repositories {
        mavenCentral()
        maven { url = 'https://s01.oss.sonatype.org/content/repositories/snapshots/' }
    }
    dependencies {
        classpath 'io.skippy:skippy-gradle:0.0.21-SNAPSHOT'
    }
}

apply plugin: io.skippy.gradle.SkippyPlugin

repositories {
    maven { url = 'https://s01.oss.sonatype.org/content/repositories/snapshots/' }
}

dependencies {
    testImplementation 'io.skippy:skippy-junit5:0.0.21-SNAPSHOT'
}
```

**Comment by fmck3516** [comment]
This has been fixed in release 0.0.22 which has been published to Maven Central and Gradle's Plugin Portal.

PRs: 
- https://github.com/skippy-io/skippy/pull/169
- https://github.com/skippy-io/skippy-regression-suite/pull/4
